src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -37,17 +37,32 @@
 #include <boost/any.hpp>
 #include <boost/range/adaptors.hpp>
 
+#if MIIR_VERSION_FLAT >= 6
+#define MIIR_BARE_POINTER_ABI 1
+#else
+#define MIIR_BARE_POINTER_ABI 0
+#endif
```

Comment:
We can deprecate this, at the soonest, after the change lands on both sides, and at the latest, after 5.4 goes out, I'd say. Because 5.4 would have the v6 ABI, which means all kernels tuned for 5.4 would be using the new ABI and so we couldn't get weird memory faults from a kernel expecting the v5 ABI being cached somewhere.